components/concept/storage/src/main/java/mozilla/components/concept/storage/BookmarksStorage.kt

```diff
@@ -133,7 +133,7 @@ data class BookmarkNode(
     val type: BookmarkNodeType,
     val guid: String,
     val parentGuid: String?,
-    val position: Int?,
+    val position: UInt?,
```

Comment:
So yeah, this _is_ the only API change. Nice!

This will propagate into  (i think) a bunch of changes in Fenix. Is it necessary to change the API here?

I guess to keep the existing API (and assuming the ffi type changed into an unsigned one), we will have to assume that we're getting a positive integer here and perform a type conversion. That sounds... okay-ish?

I'm not against churning  this and sticking with `uint` if you feel this is helpful though.

Reply:
Thanks to the stabilization of unsigned types in kotlin. This is not too much of an issue anymore! I realized I didn't link the fenix PR but it's actually an incredibly small change to get fenix compiling https://github.com/mozilla-mobile/fenix/pull/23338

However we can have some casting in this PR if you feel against this